main.py

```diff
@@ -16,39 +19,83 @@ def selfie():
+        finally:
+            cippe_db.commit()
+            return jsonify({"status": "success"}), 201
     recipe = cursor.execute("SELECT food_name FROM recipies")
     return jsonify({"status": "success", "data": recipe.fetchall()})
 
 
 @app.route("/recipies/<name>", methods=["GET"])
 def get_recipie(name):
-    cippie_db = sqlite3.connect("cippedb.db")
-    cursor = cippie_db.cursor()
+    cippe_db = sqlite3.connect("cippedb.db")
+    cursor = cippe_db.cursor()
     recipies = cursor.execute(
         "SELECT food_name, description, top_image, bottom_image FROM recipies WHERE food_name=? ORDER BY food_name ASC LIMIT 1", 
         (name, ))
     return jsonify({"status": "success", "data": recipies.fetchall()})
 
 
-@app.route("/recipies/<name>/<act>", methods=["GET"])
+@app.route("/recipies/<name>/<act>", methods=["GET", "POST"])
 def get_act(name, act):
     cippe_db = sqlite3.connect("cippedb.db")
     cursor = cippe_db.cursor()
-    key = cursor.execute("SELECT id FROM recipies WHERE food_name=?", (name, ))
+    key = cursor.execute("SELECT id FROM recipies WHERE food_name=?", (name,))
     key_id = key.fetchone()[0]
     if act == "ingredients":
-        step_ingre = cursor.execute(
-            "SELECT category, name, quantity, type, requirement, size FROM ingredients WHERE food_id=?",
+        if request.method == "POST":
+            data = request.get_json()
+            try:
+                get_ingredients(data, key_id)
+            except:
+                cippe_db.rollback()
+                return jsonify({"status": "error", "error": {"Keys(in this order)": ["category(optional)", "name(important)", "quantity(important)", "type(optional)", "requirement(optional)", "size(optional)"]}}), 404
+            finally:
+                cippe_db.commit()
+                cippe_db.close()
+        step_ingre = cippe_db.execute(
+            "SELECT category, name, quantity, type, requirement, size FROM ingredients WHERE food_id=? ORDER BY name",
             (key_id, ))
     elif act == "steps":
-        step_ingre = cursor.execute(
-            "SELECT category, name, action, image FROM steps WHERE food_id=?",(key_id, ))
+        if request.method == "POST":
+            data = request.get_json()
+            try:
+                get_steps(data, key_id)
+            except:
+                cippe_db.rollback()
+                return jsonify({"status": "error", "error": {"Keys(in this order)": ["category(optional)", "name(optional)", "action(important)", "image(optional)"]}}), 404
+            finally:
+                cippe_db.commit()
+                cippe_db.close()
+        step_ingre = cippe_db.execute(
+            "SELECT category, name, action, image FROM steps WHERE food_id=? ORDER BY action",(key_id, ))
     return jsonify({"status": "success", f"{act}": step_ingre.fetchall()})
```

Comment:
This will throw an error if `act` is  not "steps" or "ingredients". You should handle that

Reply:
Will work on this. Thanks